src/librustdoc/clean/types.rs

```diff
@@ -1114,10 +1114,7 @@ impl GenericBound {
         let path = external_path(cx, did, false, vec![], empty);
         inline::record_extern_fqn(cx, did, ItemType::Trait);
         GenericBound::TraitBound(
-            PolyTrait {
-                trait_: ResolvedPath { path, did, is_generic: false },
-                generic_params: Vec::new(),
-            },
```

Comment:
This path belongs to the `Sized` trait, so it's not an associated type, and `Path::is_assoc_ty` should preserve that because the `Res` should be "correct by construction".